### PR TITLE
Explain how to use Where to match calls with anonymous types

### LINKED ISSUE #1593: Add example of how to use A.CallTo() with anonymous types.
Having configured a fake object in strict mode, I don't seem to be able to provide a fake method call for a method that's taking an anonymous object as its parameter.

Can you please add an example to the documentation, providing information about how to deal with anonymous types in method call signatures?

### My Code (excerpt)

```c#
A.CallTo(() => _converter.ToParameters(new { loginName = "x", functionId = 1, division = (string)null }))
    .Returns(new ParameterDefinitionPowershell[0]);
```

### Exception Details

```
Call to unconfigured method of strict fake: IPowershellObjectConverter.ToParameters`1[<>f__AnonymousType1`3[System.String,System.Int32,System.String]](anonymousObject: { loginName = x, functionId = 1, division =  }).
```

### PR DESCRIPTION
**Pull request by blairconrad**
Closes #1593.

### CODE REVIEW

File: docs/argument-constraints.md [hunk 1]
```diff
@@ -1,4 +1,4 @@
-# Argument constraints
+f# Argument constraints
```
Comment (thomaslevesque) [inline docs/argument-constraints.md:1]: This seems unintentional!

**Review by thomaslevesque — APPROVED** [review]
Thanks, @blairconrad. LGTM, except for the extraneous "f" at the beginning of the file 😉

File: docs/argument-constraints.md (hunk 1 above)
Comment (blairconrad) [reply docs/argument-constraints.md:1]: What the actual F?

Glad _you_ saw it!

**Review by thomaslevesque — APPROVED** [review]
Thanks!

**Comment by blairconrad** [comment]
Thanks for the eagle-eye review and merge.

**Pull request merged**